crates/astria-proto/src/native/sequencer/v1alpha1.rs

```diff
@@ -1251,6 +1266,1158 @@ impl Display for IncorrectAddressLength {
+    #[error(
+        "the root derived from the rollup IDs in the cometbft block.data field did not match the \
+         root stored in the same block.data field"
+    )]
+    RollupIdsRootDoesNotMatchReconstructed,
+}
+
+/// A shadow of [`SequencerBlock`] with full public access to its fields.
+///
+/// This type does not guarantee any invariants and is mainly useful to get
+/// access the sequencer block's internal types.
+#[derive(Clone, Debug)]
+pub struct UncheckedSequencerBlock {
+    /// The original `CometBFT` header that was the input to this sequencer block.
+    pub header: tendermint::block::header::Header,
+    /// The collection of rollup transactions that were included in this block.
+    pub rollup_transactions: IndexMap<RollupId, Vec<Vec<u8>>>,
+    // The proof that the rollup transactions are included in the `CometBFT` block this
+    // sequencer block is derived form. This proof together with
+    // `Sha256(MTH(rollup_transactions))` must match `header.data_hash`.
+    // `MTH(rollup_transactions)` is the Merkle Tree Hash derived from the
+    // rollup transactions.
+    pub rollup_transactions_proof: merkle::Proof,
+    // The proof that the rollup IDs listed in `rollup_transactions` are included
+    // in the `CometBFT` block this sequencer block is derived form. This proof together
+    // with `Sha256(MTH(rollup_ids))` must match `header.data_hash`.
+    // `MTH(rollup_ids)` is the Merkle Tree Hash derived from the rollup IDs listed in
+    // the rollup transactions.
+    pub rollup_ids_proof: merkle::Proof,
+}
+
+/// `SequencerBlock` is constructed from a tendermint/cometbft block by
+/// converting its opaque `data` bytes into sequencer specific types.
+#[derive(Clone, Debug, PartialEq)]
+pub struct SequencerBlock {
+    /// The result of hashing `header`. Guaranteed to not be `None` as compared to
+    /// the cometbft/tendermint-rs return type.
+    block_hash: [u8; 32],
+    /// The original `CometBFT` header that was the input to this sequencer block.
+    header: tendermint::block::header::Header,
+    /// The collection of rollup transactions that were included in this block.
+    rollup_transactions: IndexMap<RollupId, Vec<Vec<u8>>>,
+    // The proof that the rollup transactions are included in the `CometBFT` block this
+    // sequencer block is derived form. This proof together with
+    // `Sha256(MTH(rollup_transactions))` must match `header.data_hash`.
+    // `MTH(rollup_transactions)` is the Merkle Tree Hash derived from the
+    // rollup transactions.
+    rollup_transactions_proof: merkle::Proof,
+    // The proof that the rollup IDs listed in `rollup_transactions` are included
+    // in the `CometBFT` block this sequencer block is derived form. This proof together
+    // with `Sha256(MTH(rollup_ids))` must match `header.data_hash`.
+    // `MTH(rollup_ids)` is the Merkle Tree Hash derived from the rollup IDs listed in
+    // the rollup transactions.
+    rollup_ids_proof: merkle::Proof,
```

Comment:
same question as on the proto - how come the proofs are here but the commitments aren't? is it because you plan to always reconstruct the `rollup_transactions_root` from `rollup_transactions`?

Reply:
Yeah, but this was always done anyway to ensure these data types uphold invariants and that the stored root and the data it is constructed from is correct.

But it need not be written to the wire format.

If it ever becomes necessary to give repeated access to this root we can add it to the idiomatic types as is done for the cometbft header block hash.